src/citrine/resources/dataset.py

```diff
@@ -273,6 +274,18 @@ def delete_contents(
                                   dataset_uid=self.uid,
                                   project_id=self.project_id
                                   )
+        if prompt_to_confirm:
+            print(f"Do you really want to delete the contents of dataset {self.uid}? [Y/N]")
+            user_response = None
+            while user_response not in {'Y', 'N'}:
+                user_response = input()
+            if user_response == 'Y':
+                pass
+            elif user_response == 'N':
+                print(f'Aborting deletion of dataset {self.uid} contents')
+                return None
```

Comment:
This is inconsistent with the return type of _poll_for_async_batch_delete_result.
```suggestion
                return []
```